### ISSUE TITLE
isolated HTML rendering

### ISSUE THREAD

**Issue by sjdv1982** [jupyter/notebook#3153]
I am trying to render arbitrary HTML inside a notebook. 
For example, the following HTML has a DOMContentLoaded event listener:

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <title>NGL - embedded</title>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
</head>
<body>
  <script src="https://cdn.rawgit.com/arose/ngl/v0.10.3/dist/ngl.js"></script>
  <script>
    var stage;
    document.addEventListener("DOMContentLoaded", function () {
      stage = new NGL.Stage("viewport");
      stage.loadFile("rcsb://1crn.mmtf", {defaultRepresentation: true});
    });
  </script>
  <div id="viewport" style="width:800px; height:800px;"> </div>
</body>
</html>
```
Rendering with %%html fails, because it needs to be in an iframe. I know the notebook supports this via the "isolated" metadata parameter, but I didn't manage to make it work like that (the parameter gets overwritten).

Right now, I am using the following workaround:

```python
html = """
<!DOCTYPE html>
...
</html>
"""
import IPython
h = IPython.display.HTML(html)
IPython.display.display_html(h,  metadata={"isolated": True}) 
```

Would it be acceptable to patch the HTML magic, so that `%%html isolated` or `%%html --isolated` will produce an iframe?

**Comment by takluyver** [comment]
I don't see a problem with adding a `--isolated` flag. The magic is in IPython, in `IPython.core.magics.display`.

**Comment by Carreau** [comment]
Closed by ipython/ipython#10962
